server/src/main/java/org/elasticsearch/action/support/ActionFilters.java

```diff
@@ -23,11 +23,16 @@
 import java.util.Comparator;
 import java.util.Set;
 
+import static java.util.Collections.emptySet;
+
 /**
  * Holds the action filters injected through plugins, properly sorted by {@link org.elasticsearch.action.support.ActionFilter#order()}
  */
 public class ActionFilters {
 
+    // this could be used in many more places - TODO use this where appropriate
+    public static final ActionFilters EMPTY_FILTERS = new ActionFilters(emptySet());
```

Comment:
I'm not sure this is a good idea. This object should only be manually created by tests. In production code, the ActionFilters instance comes from the system. By putting this here, it might give you the impression that it's ok to directly use this instance instead.

Reply:
Ok, backed this out.